Framework/Algorithms/src/SmoothNeighbours.cpp

```diff
@@ -735,7 +730,7 @@ void SmoothNeighbours::spreadPixels(MatrixWorkspace_sptr outws) {
   for (int outWIi = 0; outWIi < int(numberOfSpectra); outWIi++) {
     const auto &inSpec = inWS->getSpectrum(outWIi);
     auto &outSpec2 = outws2->getSpectrum(outWIi);
-    outSpec2.dataX() = inSpec.dataX();
+    outSpec2.mutableX() = inSpec.x();
     outSpec2.addDetectorIDs(inSpec.getDetectorIDs());
     // Zero the Y and E vectors
     outSpec2.clearData();
```

Comment:
`clearData` sets Y and E to zero, so either remove this (and change dataY to mutableY) or remove the two lines below. Also note that we have `operator=(double)`, i.e., if you need it you can write

```cpp
outputSpec2.mutableY() = 0.0;
```